from_dict/_from_dict.py

```diff
@@ -138,9 +139,36 @@ def get_constructor_type_hints(
     if cls is None:
         return {}
 
-    hints = typing.get_type_hints(
-        cls.__init__, ns_types.global_types, ns_types.local_types
-    ) or typing.get_type_hints(cls, ns_types.global_types, ns_types.local_types)
+    get_type_hints = functools.partial(
+        typing.get_type_hints,
+        globalns=ns_types.global_types,
+        localns=ns_types.local_types,
+    )
+
+    if hasattr(cls, "__parameters__"):
```

Comment:
This is the simplest approach I found to determine if a cls is generic. 
According to the Python [Documenation](https://docs.python.org/3.9/library/stdtypes.html?highlight=__parameters__#genericalias.__parameters__) Generic classes will always have a `__parameters__` field

Below is another approach to determining if a class is generic. I decided against this one because it seemed unnecessarily  complex.
```python
C = TypeVar("C")
def is_generic_type(
    cls: Type,
) -> bool:
    # The type class is private
    # from typing import _GenericAlias
    #return isinstance(cls, _GenericAlias)
    GENERIC_TYPE = type(Generic[C])
    return isinstance(cls, GENERIC_TYPE)

```

Reply:
While writing this comment I found some information that makes me thing the second approach I mentioned is better. 🤷

Reply:
After some testing I changed my mind again. :confounded: Anything other than `hasattr(cls, "__parameters__")` has inconsistent behavior across Python versions.